ow_plexil/src/plexil-adapter/PlexilPlanSelection.cpp

```diff
@@ -84,40 +84,58 @@ void PlexilPlanSelection::waitForPlan(){
 void PlexilPlanSelection::runCurrentPlan(){
   std_msgs::String status;
   ros::Rate rate(10); // 10 Hz for overall rate we are spinning
-  //try to run the plan
-  if(OwExecutive::instance()->runPlan(plan_array[0].c_str())){
-    //workaround for getPlanState not working on first plan
-    if(m_first_plan == true){
-      m_first_plan = false;
-    }
+
+  if (OwExecutive::instance()->runPlan(plan_array[0].c_str())) {
+
+/* Skipping these checks for now, because they don't work if the plan
+   finishes execution very quickly.  Seeking a solution from the
+   PLEXIL team as of 12/16/22.
+
+    // Workaround for allPlansFinished() not working on first plan.
+    if (m_first_plan) m_first_plan = false;
+
     // Times out after 3 seconds or the plan is registered as running.
     int timeout = 0;
-    while(OwExecutive::instance()->getPlanState() && timeout < 30){
+    while(OwExecutive::instance()->allPlansFinished() && timeout < 30){
```

Comment:
There are a lot of literals here that require doing math to realize their origin, so I suggest defining the constants `float LOOP_RATE = 10` and `float TIMEOUT = 3.0`
```suggestion
    while(OwExecutive::instance()->allPlansFinished() && timeout < static_cast<int>(LOOP_RATE * TIMEOUT)){
```

Reply:
I just realized this is all commented out. Feel free to ignore this suggestion if this code is slated for removal.

Reply:
Good suggestion.  Adding todo for future work, which I think is best done when the commented-out code is replaced with a better solution that I'm working on with the Plexil team.

Reply:
Good suggestion, done.